x-pack/plugins/observability/server/services/slo/summary_cleanup_task.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger, SavedObjectsClient } from '@kbn/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { SortResults } from '@elastic/elasticsearch/lib/api/types';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../assets/constants';
+import { StoredSLO } from '../../domain/models';
+import { SO_SLO_TYPE } from '../../saved_objects';
+
+export const TASK_TYPE = 'SLO-DEFINITIONS-CLEANUP-TASK';
+
+function findMissingItems(soIds: string[], sloDefIds: string[]) {
+  const missingSloDefIds = sloDefIds.filter((item) => !soIds.includes(item));
+  const missingSOIds = soIds.filter((item) => !sloDefIds.includes(item));
+
+  return {
+    missingSloDefIds,
+    missingSOIds,
+  };
+}
+
+export class SloSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClient;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '1m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              const runAt = new Date().toISOString();
+
+              if (this.soClient && this.esClient) {
+                const finder = this.soClient.createPointInTimeFinder<StoredSLO>({
```

Comment:
I guess technically we don't have a StoredSLO but a `Pick<StoredSLO, "id">` (and revision when added in the fields list)